mmdet/evaluation/metrics/voc_metric.py

```diff
@@ -121,19 +121,7 @@ def compute_metrics(self, results: list) -> dict:
         eval_results = OrderedDict()
         if self.metric == 'mAP':
             assert isinstance(self.iou_thrs, list)
-            dataset_type = self.dataset_meta.get('DATASET_TYPE')
-            if dataset_type in ['VOC2007', 'VOC2012']:
```

Comment:
Add a todo list here, we need find a more elegant way, or even fix the potential bug in mmengine later